### PR TITLE
Fix torch Conv2DTranspose with padding='same' and stride > 1

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Description

The torch backend's `conv_transpose` produced values shifted by one pixel for `padding='same'` with `strides > 1` on odd kernels. Output shapes were correct so this slipped past CI, but the actual values diverged from JAX/TensorFlow by O(magnitude of input) for any real workload.

**Repro on master:**

```python
import os, numpy as np
os.environ["KERAS_BACKEND"] = "torch"  # try also "jax", "tensorflow"
import keras

rng = np.random.default_rng(42)
x = rng.standard_normal((4, 16, 16, 16)).astype("float32")
kernel = (rng.standard_normal((3, 3, 32, 16)) * 0.05).astype("float32")

layer = keras.layers.Conv2DTranspose(
    filters=32, kernel_size=3, strides=2, padding="same", use_bias=False
)
layer.build(x.shape)
layer.set_weights([kernel])
out = np.asarray(keras.ops.convert_to_numpy(layer(x)))
print(out.sum())
# torch  : 16.10
# jax    : 44.60
# tf     : 44.60
```

Sweep across all four `(padding, stride)` combinations:

| padding | stride | jax ↔ torch max\|Δ\| before | after |
|---|---|---|---|
| valid | 1 | 1.2e-6 | 1.2e-6 |
| valid | 2 | 5e-7 | 5e-7 |
| same | 1 | 1.2e-6 | 1.2e-6 |
| **same** | **2** | **2.23** | **5e-7** |

**Root cause:** `torch.nn.functional.conv_transpose*d` takes a symmetric `padding` plus right-only `output_padding`. Keras "same" with `stride > 1` and odd kernel requires asymmetric padding: for `stride=2, kernel=3`, JAX computes `(left_pad=2, right_pad=1)`, which the existing torch mapping at `_convert_conv_transpose_padding_args_from_keras_to_torch` replaced with `(padding=1, output_padding=1)` — same output shape, wrong spatial alignment. The helper acknowledged this with `warnings.warn(...)` when both values are positive, and the test suite documented it with `with pytest.raises(AssertionError): ...`, but neither prevented users from getting silently wrong outputs.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/backend/torch/nn.py [hunk 1]
```diff
@@ -720,55 +720,56 @@ def conv_transpose(
...
     kernel = _transpose_conv_kernel(kernel)
 
-    if data_format == "channels_last":
-        inputs = _maybe_convert_to_channels_last(inputs)
-        kernel = _maybe_convert_to_channels_last(kernel)
-
-    kernel_spatial_shape = kernel.shape[2:]
     if isinstance(dilation_rate, int):
-        dilation_rate = [dilation_rate] * len(kernel_spatial_shape)
+        dilation_rate = [dilation_rate] * num_spatial_dims
```
[the remark this answers is not in the document]
Comment (MarcosAsh) [reply keras/src/backend/torch/nn.py:748]: The new path slices the conv output per spatial dim, which produces a non-contiguous view. The  channels_last memory format would be invalidated by the slice anyway and would force an extra copy to restore.

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!

**Comment by MarcosAsh** [comment]
Hi, thanks for the review! I fixed the merge conflict.

**Review by hertschuh — APPROVED** [review]
Thanks!
